packages/x-date-pickers/src/PickersDay/PickersDay.tsx

```diff
@@ -342,8 +290,9 @@ const PickersDayRaw = React.forwardRef(function PickersDay<TDate>(
       data-mui-test="day"
       disabled={disabled}
       tabIndex={selected ? 0 : -1}
-      onFocus={handleFocus}
-      onKeyDown={handleKeyDown}
+      onKeyDown={(event) => onKeyDown(event, day)}
```

Comment:
You can avoid the `noop` with `onKeyDown?.(event, day)`

But as you want

Reply:
Already did a lot of modification, I prefer to stick with what was there before